extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,801 @@
+      await overrideQueryHelp(variantAnalysis, cliServer, localAutofixPath);
+
+      // Get the full names (nwos) of the selected repositories.
+      const selectedRepoNames = getSelectedRepositoryNames(
+        variantAnalysis,
+        filterSort,
+      );
+
+      // Get storage paths for the autofix results.
+      const {
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+      } = await getStoragePaths(variantAnalysisId, storagePath);
+
+      // Process the selected repositories:
+      //  Get sarif
+      //  Download source root
+      //  Run autofix and output results
+      progress(
+        progressUpdate(
+          3,
+          4,
+          `Processing ${pluralize(selectedRepoNames.length, "repository", "repositories")}`,
+        ),
+      );
+      const outputTextFiles = await processSelectedRepositories(
+        selectedRepoNames,
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+        localAutofixPath,
+        credentials,
+        logger,
+      );
+
+      // Output results from all repos to a combined markdown file.
+      progress(progressUpdate(4, 4, `Finalizing autofix results`));
+      const combinedOutputMarkdownFile = join(
+        autofixOutputStoragePath,
+        "autofix-output.md",
+      );
+      await mergeFiles(outputTextFiles, combinedOutputMarkdownFile, false);
+
+      // Open the combined markdown file.
+      await tryOpenExternalFile(app.commands, combinedOutputMarkdownFile);
+    },
+    {
+      title: "Generating Autofixes",
+      cancellable: false, // not cancellable for now
+    },
+  );
+}
+
+/**
+ * Finds the local autofix installation path from the AUTOFIX_PATH environment variable.
+ * Throws an error if the path is not set or does not exist.
+ * @returns An object containing the local autofix path.
+ * @throws Error if the AUTOFIX_PATH environment variable is not set or the path does not exist.
+ */
+function findLocalAutofix(): string {
+  const localAutofixPath = process.env.AUTOFIX_PATH;
+  if (!localAutofixPath) {
+    throw new Error("Path to local autofix installation not found.");
+  }
+  if (!pathExists(localAutofixPath)) {
+    throw new Error(`Local autofix path ${localAutofixPath} does not exist.`);
+  }
+  return localAutofixPath;
+}
+
+/**
+ * Overrides the query help from a given variant analysis
+ * at a location within the `localAutofixPath` directory .
+ */
+async function overrideQueryHelp(
+  variantAnalysis: VariantAnalysis,
+  cliServer: CodeQLCliServer,
+  localAutofixPath: string,
+): Promise<void> {
+  // Get path to the query used by the variant analysis.
+  const queryFilePath = variantAnalysis.query.filePath;
+  if (!(await pathExists(queryFilePath))) {
+    throw new Error(`Query file used by variant analysis not found.`);
+  }
+  const queryFilePathNoExt = join(
+    dirname(queryFilePath),
+    parse(queryFilePath).name,
```

Comment:
It seems like `parse` also returns a `dir`, so to avoid calling multiple `path` functions, could we store the result of `parse` in a variable and then do `join(parsedQueryFilePath.dir, parsedQueryFilePath.name)`?